tools/management/commands/upload_excel_bias.py

```diff
@@ -583,3 +619,36 @@ def prepare_all_data(self, filenames):
                 self.data_all += rows
         print(len(self.data_all), " total data points")
         print("Finished")
+
+    def create_empty_ligand(self, ligand_name):
+        # gtoplig webresource
+        lp = self.build_ligand_properties()
+        ligand = Ligand()
+        ligand.properities = lp
+        ligand.name = ligand_name
+        ligand.canonical = True
+        ligand.ambigious_alias = False
+        ligand.pdbe = None
+        try:
+            ligand.save()
+        except IntegrityError:
+            return Ligand.objects.get(name=ligand_name, canonical=True)
+        return ligand
+        # return self.update_ligand(ligand_name, {}, ligand_type, web_resource, gtop_id)
+
+# pylint: disable=C0301
+# pylint: disable=R0201
+    def build_ligand_properties(self):
+        lp = LigandProperities()
+        lt =  LigandType.objects.filter(name = 'small molecule')[0]
```

Comment:
It's better to do a get instead of a filter when you are requesting a single object. Don't forget to add try/catch for error handling.

Reply:
this func is called when getting any ligands data from pubchem or gpcrdb database resulted in none. 
In the matter to save the annotated ligands, null ligand properties are created

Reply:
I understand, but at line 643 you request a ligandType object for "small molecule" and expect one instance back. In this case it's better to do a get call instead of a filter call.

Reply:
Done